mne/encoding/model.py

```diff
@@ -1,28 +1,66 @@
 import numpy as np
 from sklearn.base import BaseEstimator
 from scipy import linalg
+from .feature import EventsBinarizer, DataDelayer
 from ..externals.six import string_types
+from ..utils import warn
+from sklearn.base import is_regressor
+
+
+class EventRelatedRegressor(object):
+    def __init__(self, raw, events, est=None, event_id=None, tmin=-.5, tmax=.5,
+                 preproc_x=None, preproc_y=None):
+        if events.shape[-1] != 3:
+            raise ValueError('Events must be shape (n_events, 3)')
+        if raw.preloaded is False:
+            raise ValueError('Data must be preloaded')
+
+        # Create events representation
+        self.ev_ixs = events[:, 0]
+        self.ev_types = events[:, 2]
+        self.event_id = event_id
+        binarizer = EventsBinarizer(raw.n_times, sfreq=raw.info['sfreq'])
+        self.X = binarizer.fit_transform(self.ev_ixs, self.ev_types,
+                                         self.event_id)
+
+        # Prep output data
+        self.raw = raw
+
+        # Prepare data preprocessors + design matrix
+        delayer = DataDelayer(time_window=[tmin, tmax],
+                              sfreq=sig_cont.info['sfreq'])
+        _check_preproc(preproc_x)
+        _check_preproc(preproc_y)
+        if preproc_x is not None:
+            # Add the delays to the end of preproc_x
+            preproc_x = Pipeline(preproc_x.steps + [('delayer', DataDelayer)])
+
+        # Create model and attributes
+        self.est = EncodingModel(est, preproc_x=preproc_x, preproc_y=preproc_y)
+        self.tmin = tmin
+        self.tmax = tmax
+
+    def fit(self, picks=None):
+        if picks is None:
+            picks = np.arange(len(self.raw.ch_names))
+        Y = self.raw._data[picks]
+        self.est.fit(self.X, Y)
+        self.coef_ = self.est.coef_
```

Comment:
no, it'd just be for convenience. Think we can scrap it? I agree that it's better not to duplicate the variable but didn't want people getting confused and having to dig for their variables. Actually now that I think about it, since we're using the estimator's `predict` method, this would sort of remove the requirement that the estimator have a `coef_` attribute